lib/discard.js

```diff
@@ -18,7 +18,7 @@ module.exports = function (state, callback) {
         tmpPath = state.template.tmp;
 
     // Remove the .git directory unless told otherwise
-    if (isDirectory.sync(tmpPath + '/.git') && !state.data.keepHistory && !(state.flags && state.flags['keep-history'])) {
+    if (isDirectory.sync(tmpPath + '/.git') && !state.data.keepHistory && (!state.flags || !state.flags['keep-history'])) {
```

Comment:
Not sure this was necessary :P Just kinda ended up here, but I think it works anyway :)